framework/src/transfers/MultiAppUserObjectTransfer.C

```diff
@@ -389,109 +395,108 @@ MultiAppUserObjectTransfer::execute()
         }
       }
 
-      for (unsigned int i = 0; i < _multi_app->numGlobalApps(); i++)
+      if (is_nodal)
       {
-        if (!_multi_app->hasLocalApp(i))
-          continue;
+        for (auto & node : to_mesh->getMesh().node_ptr_range())
+        {
+          if (blockRestricted() && !hasBlocks(to_mesh, node))
+            continue;
 
-        Point app_position = _multi_app->position(i);
-        BoundingBox app_box = _multi_app->getBoundingBox(i, _displaced_source_mesh);
-        const UserObject & user_object = _multi_app->appUserObjectBase(i, _user_object_name);
+          if (boundaryRestricted() && !isBoundaryNode(to_mesh, node))
+            continue;
 
-        if (is_nodal)
-        {
-          for (auto & node : to_mesh->getMesh().node_ptr_range())
+          if (node->n_dofs(to_sys_num, to_var_num) > 0) // If this variable has dofs at this node
           {
-            if (blockRestricted() && !hasBlocks(to_mesh, node))
-              continue;
+            const auto sub_app = findSubAppToTransferFrom(*node);
 
-            if (boundaryRestricted() && !isBoundaryNode(to_mesh, node))
+            // Check to see if a sub-app was found
+            if (sub_app == static_cast<unsigned int>(-1))
```

Comment:
is there a check that at least one subapp was found somewhere?

Reply:
It's guaranteed, by the algorithm, to find one.